Gem/Code/Source/MeshExampleComponent.cpp

```diff
@@ -98,24 +98,68 @@ namespace AtomSampleViewer
         m_lowEndPipeline = nullptr;
     }
 
+    void MeshExampleComponent::CreateDeferredPipeline()
+    {
+        AZ::RPI::RenderPipelineDescriptor pipelineDesc;
+        pipelineDesc.m_mainViewTagName = "MainCamera";
+        pipelineDesc.m_name = "DeferredPipeline";
+        pipelineDesc.m_rootPassTemplate = "DeferredPipelineTemplate";
+        pipelineDesc.m_renderSettings.m_multisampleState.m_samples = 4;
+        pipelineDesc.m_allowModification = true; // MainPipeline allows modifications, so the DeferredPipeline must as well, to get a consistent result.
+
+        m_deferredPipeline = AZ::RPI::RenderPipeline::CreateRenderPipelineForWindow(pipelineDesc, *m_windowContext);
+    }
+
+    void MeshExampleComponent::DestroyDeferredPipeline()
+    {
+        m_deferredPipeline = nullptr;
+    }
+
     void MeshExampleComponent::ActivateLowEndPipeline()
     {
-        m_originalPipeline = m_scene->GetDefaultRenderPipeline();
+        AZ::RPI::RenderPipelinePtr prevPipeline = m_scene->GetDefaultRenderPipeline();
+
+        if (!m_originalPipeline)
+        {
+            m_originalPipeline = prevPipeline;
+        }
+
         m_lowEndPipeline->GetRootPass()->SetEnabled(true); // PassSystem::RemoveRenderPipeline was calling SetEnabled(false)
         m_scene->AddRenderPipeline(m_lowEndPipeline);
-        m_lowEndPipeline->SetDefaultView(m_originalPipeline->GetDefaultView());
-        m_scene->RemoveRenderPipeline(m_originalPipeline->GetId());
+        m_lowEndPipeline->SetDefaultView(prevPipeline->GetDefaultView());
+        m_scene->RemoveRenderPipeline(prevPipeline->GetId());
 
+        m_imguiScope = {}; // I'm not sure why this is needed. Something must be wrong in the ImGuiActiveContextScope class
         m_imguiScope = AZ::Render::ImGuiActiveContextScope::FromPass({ m_lowEndPipeline->GetId().GetCStr(), "ImGuiPass" });
     }
 
-    void MeshExampleComponent::DeactivateLowEndPipeline()
+    void MeshExampleComponent::ActivateDeferredPipeline()
```

Comment:
Suggest an `ActivatePipeline` that accepts an argument to the pipeline to activate

Reply:
At some point I plan to move these controls to a sharable location, where other Samples can include the same functionality. I'll see if I can clean up the controls at that time.